Phos refactor

ErinCain opened the pull request:
PHOS update draft

ErinCain committed:
Updates model logic to track hatch and natural seperatly

ErinCain committed:
updates hatch dynamics

ErinCain committed:
fixes tibble warning

ErinCain committed:
Adds potential todo for updating

ErinCain committed:
first attempt at phos renaturing logic

ErinCain committed:
removes debug code and updates phos logic

ErinCain committed:
Updates docs for hatchery update

ErinCain committed:
Adds comments and cache for storing limited hab analysis

ErinCain committed:
adds context to hatchery markdown

ErinCain committed:
Adds comma in after new data object output

ErinCain committed:
Initiate realease data analysis

ErinCain committed:
Adds hatch documenation and updates release analysis

ErinCain committed:
Add todos for updating model logic

ErinCain committed:
updates hatchery release logic, create hatchery release data object

ErinCain committed:
add back in chipps releases:

ErinCain committed:
Updates NA handeling and gets running with hatch logic

ErinCain committed:
Update return dist for hatchery adults

ErinCain committed:
shifts position of juv logic to capture month

ErinCain committed:
Adds notes from walkthrough

ergz committed:
moves renature calculation so that they get executed only code part that is affected

ergz committed:
indents some stuff and moves sim_length to top

ergz committed:
Merge branch 'phos-refactor' of github.com:Reorienting-to-Recovery/fallRunDSM into phos-refactor

# Conflicts:
#	R/model.R

ergz commented on R/model.R:
```diff
@@ -183,14 +236,28 @@ fall_run_model <- function(scenario = NULL, mode = c("seed", "simulate", "calibr
+    # R2R hatchery logic -------------------------------------------------------
+    # Currently adds only on major hatchery rivers (American, Battle, Feather, Merced, Moke)
+    # Add all as fry? or should we do larger
+    natural_juveniles <- rowSums(juveniles * natural_proportion_with_renat)
```
simple optimization: multiplying each col then adding is the same as adding them all then multiplying

```r
total_juves_pre_hatchery <- rowSums(juveniles)
natural_juveniles <- total_juves_pre_hatchery  * natural_proportion_with_renat
```

ErinCain replied:
updated!

ergz commented on R/model.R:
```diff
@@ -658,49 +724,80 @@ fall_run_model <- function(scenario = NULL, mode = c("seed", "simulate", "calibr
+    })) * output$proportion_natural_juves_in_tribs[ , year]
+
+    # TODO see if better place to do this
+    natural_adults_returning[is.na(natural_adults_returning)] = 0
```
i think you should switch to `is.nan`, NaN's are ok to appear in the code at this point (and be replaced) its just the places where denominator is 0, but if NA appears then something went wrong in the code you don't want to just replace those with zeros they should propagate and be fixed when they appear.

ErinCain replied:
updated

ErinCain committed:
updates replace Na with NaN instead of 0

ErinCain committed:
Updates logic to increase performance, removes todos

ErinCain committed:
Merge branch 'r2r-main' into phos-refactor

ErinCain committed:
cleans up todos

ErinCain committed:
Updates logic for hatchery releases at chips so that they experience ocean entry success

ErinCain committed:
moves hatchery release logic into cache data and updates params accordingly

ErinCain committed:
Inits params for no hatchery

ErinCain committed:
saves file for inits params for no hatchery

ErinCain committed:
moves model parameter scripts into directory

ErinCain committed:
initiates no harvest parameters

ErinCain committed:
updates documenation on hatchery update

ErinCain committed:
updates max hab params to have hatch release

ErinCain committed:
inits max flow params

ErinCain committed:
fixes error in flow data, updates max flow params

ErinCain committed:
Adds max flow max hab params

ErinCain committed:
Adds proxy max hatchery run

ErinCain committed:
Update params lists with minor mods

ErinCain committed:
Adds updated max hatch params

ErinCain committed:
Updates releases to production targets

ErinCain committed:
Update prey density to medium

ErinCain committed:
Adds in analysis and data object for harvest

ErinCain committed:
reruns baseline params with new prey density, updated hatchery numbers, and updated harvest rate

ErinCain committed:
Updates model logic to include fecundity variation by age, adds fecundity data object to params

ErinCain committed:
Adds package call for non base functions, adds todos, adds conditional with default rations when total fish in trib is 0 so that straying fish can still spawn'

ErinCain commented:
Updated and removed draft status of PR - updates include: 
- updated hatchery release numbers 
- updated harvest rate 
- updated fecundity based on age (and associated model logic) 
- updated bookend params with new data objects ---TODO---still need the spawn decay multiplier in there. @ergz can you add this once you have rerun the spawn decay script in the R2R/DSMhabitat
- Initiated params for all the bookend scenarios __do not worry about reviewing these now because I will update all of them after the baseline calibration___

The results still seem reasonable but the new fecundity logic could definitely use a thorough review. I added some remaining TODOs in there for when we have time. It would be good to start functionalizing some of this R2R stuff where it it makes sense to clean up the script but I think that can wait until after this SDM call when we hopefully have a bit more time. 

Happy to hop on a quick call to discuss changes.

ErinCain committed:
Adds comment to cache data to refer to location of R2R data objects

ErinCain committed:
Update name of return_hatch to hatch_adults

ErinCain committed:
adds updated baseline params with spawn decay

ergz committed:
adds default params for r_to_r in main model run

ErinCain committed:
pulls in calibration of baseline run

ErinCain committed:
Updates model parameters for bookends with new calibration

ErinCain committed:
pull origin

ErinCain committed:
Saves updates to tmh params

ErinCain committed:
updates calculated_adults handeling for calibration

ErinCain committed:
updates to newest calibration

ErinCain committed:
Adds fix for bear river rear surv intercept and updates to newest calibration

ErinCain committed:
Saves param updates

ErinCain committed:
updates params for new calibration

ErinCain committed:
Adds max hab params and stops neg spawners

ErinCain committed:
Updates dcc gates logic for Run of River flows in max flow params

ErinCain committed:
Updates max flow params to fix missing delta hab

ErinCain committed:
Reruns all fall run params with new spawn decay multiplier

ErinCain committed:
updates no hathch params with new spawn decay multiplier

ErinCain committed:
Adds updated max flow prop pulse to remove butte NaN

ErinCain committed:
Updates model params
